.gitlab/dev_container_deploy/docker_linux.yml

```diff
@@ -224,13 +224,14 @@ single_machine_performance-nightly-amd64-a7:
   extends: .docker_publish_job_definition
   stage: dev_container_deploy
   rules:
-    !reference [.on_deploy_nightly_repo_branch_a7]
+    - !reference [.on_deploy_nightly_repo_branch_a7]
+    - !reference [.if_main_branch]
```

Comment:
I’m not a GitLab expert, so maybe @DataDog/agent-platform can confirm/infirm but I’m not sure that this will have the expected behavior.

If I understand correctly, the goal is to have this job triggered only if *“it’s a nightly build"* **AND** *“it’s on `main` branch”*.
But [the semantic of `rules` list](https://docs.gitlab.com/ee/ci/yaml/#rules) is:
> Rules are […] evaluated in order until the first match. When a match is found, the job is either included or excluded from the pipeline, depending on the configuration.

So, I fear that this will trigger the job if *“it’s a nightly build"* **OR** *“it’s on `main` branch”*.

Reply:
As Lenaic said, this is an `OR` (Gitlab matches each rule successively, not in conjunction).

In this case, since the goal is to run this job only on nightly (ie. scheduled) pipelines on `main`, the existing `if_scheduled_main` rule (defined [here](https://github.com/DataDog/datadog-agent/blob/99e849b58997f41adaeaab6b0b92d0e3c08e6a66/.gitlab-ci.yml#L286-L288)) should do what you need.